pkg/master/controller.go

```diff
@@ -155,15 +174,14 @@ func (c *Controller) CreateMasterServiceIfNeeded(serviceName string, serviceIP n
 			Labels:    map[string]string{"provider": "kubernetes", "component": "apiserver"},
 		},
 		Spec: api.ServiceSpec{
-			Ports: []api.ServicePort{{Port: servicePort, Protocol: api.ProtocolTCP, TargetPort: util.NewIntOrStringFromInt(servicePort)}},
+			Ports: ports,
 			// maintained by this code, not by the pod selector
 			Selector:        nil,
 			ClusterIP:       serviceIP.String(),
```

Comment:
I guess by incorrect, you are saying its unnecessary... I think i see your point.  IF so, then I would say that this semantic be reflected in the api types, if possible.  but i will dive a little into this tomorrow.  its a good catch.